src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -0,0 +1,323 @@
+                isBackflow ? "VMR" : mapping.Name,
+                lastFlow.SourceSha,
+                shaToFlow);
+
+            if (discardPatches)
+            {
+                foreach (VmrIngestionPatch patch in patches)
+                {
+                    _fileSystem.DeleteFile(patch.Path);
+                }
+            }
+
+            return null;
+        }
+
+        _logger.LogInformation("Created {count} patch(es)", patches.Count);
+
+        await _localGitClient.CheckoutAsync(targetRepo, lastFlow.TargetSha);
+        await _workBranchFactory.CreateWorkBranchAsync(targetRepo, branchName);
+
+
+        // TODO: Remove VMR patches before we create the patches
+
+        try
+        {
+            foreach (VmrIngestionPatch patch in patches)
+            {
+                await _vmrPatchHandler.ApplyPatch(patch, targetRepo, cancellationToken);
+
+                if (discardPatches)
+                {
+                    _fileSystem.DeleteFile(patch.Path);
+                }
+            }
+        }
+        catch (Exception e) when (e.Message.Contains("Failed to apply the patch"))
+        {
+            // TODO: This can happen when we also update a PR branch but there are conflicting changes inside. In this case, we should just stop. We need a flag for that.
+
+            // This happens when a conflicting change was made in the last backflow PR (before merging)
+            // The scenario is described here: https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Full-Code-Flow.md#conflicts
+            _logger.LogInformation("Failed to create PR branch because of a conflict. Re-creating the previous flow..");
+
+            // Find the last target commit in the repo
+            var previousRepoSha = await BlameLineAsync(
+                repoPath / VersionFiles.VersionDetailsXml,
+                line => line.Contains(VersionDetailsParser.SourceElementName) && line.Contains(lastFlow.SourceSha),
+                lastFlow.TargetSha);
+            await _localGitClient.CheckoutAsync(targetRepo, previousRepoSha);
+
+            // Reconstruct the previous flow's branch
+            branchName = await FlowCodeAsync(
+                isBackflow: true,
+                repoPath,
+                mapping.Name,
+                lastFlow.SourceSha,
+                discardPatches,
+                cancellationToken);
+
+            // The current patches should apply now
+            foreach (VmrIngestionPatch patch in patches)
+            {
+                // TODO: Catch exceptions?
+                await _vmrPatchHandler.ApplyPatch(patch, targetRepo, cancellationToken);
+
+                if (discardPatches)
+                {
+                    _fileSystem.DeleteFile(patch.Path);
+                }
+            }
+        }
+
+        var commitMessage = $"""
+            [{(isBackflow ? "VMR" : mapping.Name)}] Codeflow {shortShas}
```

Comment:
Another place where `{(isBackflow ? "VMR" : mapping.Name)}` would be benefitial to have as a method of the code flow (presuming it is enriched by information about the repo and vmr, not just the shas).

Reply:
Removed, it's actually obvious this will be always true